folly/io/async/AsyncSocket.cpp

```diff
@@ -662,7 +662,11 @@ void AsyncSocket::writeChain(WriteCallback* callback, unique_ptr<IOBuf>&& buf,
   constexpr size_t kSmallSizeMax = 64;
   size_t count = buf->countChainElements();
   if (count <= kSmallSizeMax) {
+    // suppress "warning: variable length array ‘vec’ is used [-Wvla]"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wvla"
```

Comment:
If you are trying to build without VLAs, what about unsetting `FOLLY_HAVE_VLA` instead?